source/iface/common-reference.rst

```diff
@@ -4,30 +4,162 @@
+* ``T`` must be equality comparable in the host application. Equality between two 
+  instances of T (i.e. ``a == b``) must be true if one instance is a copy of the 
+  other and non-equality between two instances of ``T`` (i.e. ``a != b``) must 
+  be true if neither instance is a copy of the other, in accordance with the 
+  above definition of a copy, unless either instance has become invalidated 
+  by a move operation. By extension of the requirements above, equality on ``T`` 
+  must guarantee to be reflexive (i.e. ``a == a``), symmetric (i.e. ``a == b`` 
+  implies ``b == a`` and ``a != b`` implies ``b != a``) and 
+  transitive (i.e. ``a == b && b == c`` implies ``c == a``).
+* A specialization of ``std::hash`` for ``T`` must exist on the host application 
+  that returns a unique value such that if two instances of T are equal, in 
+  accordance with the above definition, then their resulting hash values are 
+  also equal and subsequently if two hash values are not equal, then their 
+  corresponding instances are also not equal, in accordance with the above definition.
 
-Classes with reference semantics support the following methods. An
-instance that is constructed as a copy of another instance must behave
-as-if it were the same instance.
+Some SYCL runtime classes will have additional behavior associated with copy, 
+movement, assignment or destruction semantics. If these are specified they are 
+in addition to those specified above unless stated otherwise.
+
+Each of the runtime classes mentioned above must provide a common interface 
+of special member functions in order to fulfill the copy, move, destruction
+requirements and hidden friend functions in order to fulfill the equality 
+requirements.
 
 ::
 
+  namespace sycl {
+
   class T {
-  public:
-    T(const T &rhs);
-    T(T &&rhs);
-    T &operator=(const T &rhs);
-    T &operator=(T &&rhs);
+    ...
+
+    public: 
+
+    T(const T& rhs);
+
+    T(T&& rhs);
+
+    T& operator=(const T& rhs);
+
+    T& operator=(T&& rhs);
+
     ~T();
-    friend bool operator==(const T &lhs, const T &rhs) { /* ... */ }
-    friend bool operator!=(const T &lhs, const T &rhs) { /* ... */ }
+
+    ...
+
+    friend bool operator==(const T& lhs, const T& rhs) { /* ... */ }
+
+    friend bool operator!=(const T& lhs, const T& rhs) { /* ... */ }
+
+    ...
   };
+
+  } // namespace sycl
+
+.. seealso:: |SYCL_SPEC_COMMON_REFERENCE|
+
+.. _ref_special_member_func:
+
+=======================================================
+Common special member functions for reference semantics
+=======================================================
+
+.. list-table::
+  :header-rows: 1
+
+  * - Special member function
+    - Description
+  * - ``T(const T& rhs);``
+    - Constructs a ``T`` instance as a copy of the RHS SYCL 
+      ``T`` in accordance with the requirements set out above.
+  * - ``T(T&& rhs);``
+    - Constructs a ``T`` instance as a copy of the RHS SYCL 
```

Comment:
```suggestion
    - Constructs a SYCL ``T`` instance as a copy of the RHS SYCL 
```

Reply:
Fixed in https://github.com/KhronosGroup/SYCL_Reference/pull/92/commits/bee24d7331f20616d586c618e1be1b7b4580bcb6.